packages/astroport/src/pair.rs

```diff
@@ -170,3 +170,15 @@ pub enum StablePoolUpdateParams {
     StartChangingAmp { next_amp: u64, next_amp_time: u64 },
     StopChangingAmp {},
 }
+
+pub fn migration_check(deps: Deps, factory: Addr, pair_addr: Addr) -> StdResult<bool> {
```

Comment:
Seems it's enough to use `factory: &Addr, pair_addr: &Addr`. So no need to clone addresses every time.